karapace/client.py

```diff
@@ -0,0 +1,179 @@
+"""
+karapace - utils
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from aiohttp import ClientSession
+from collections.abc import Mapping
+from typing import Awaitable, Callable, Optional, Union
+from urllib.parse import urljoin
+
+import logging
+import ssl
+
+log = logging.getLogger(__name__)
+Path = str
+Headers = dict
+JsonData = object  # Type of the result after parsing JSON
+
+
+async def _get_aiohttp_client() -> ClientSession:
```

Comment:
This only exists because the function should be async, but it is not used outside of this module, so I made it private. IMO it would make sense to remove this callback and just inject the `ClientSession` in the `Client` constructor.